src/Components/Form/Form.js

```diff
@@ -1,154 +1,243 @@
+import "./Form.scss";
 import { RiHomeSmileLine } from "react-icons/ri";
-import { Link } from 'react-router-dom';
+import { Link, Redirect } from "react-router-dom";
 import { CgUserlane } from "react-icons/cg";
 export class Form extends Component {
   constructor(props) {
-    super(props)
+    super(props);
     this.state = {
+      history:[],
       prompts: [],
       allAnswers: [],
       currentAnswers: [],
       questionsPerActivity: [],
-    }
+    };
   }
 
   componentDidMount = async () => {
-    this.setState({prompts: await getAllQuestions()})
+    this.setState({ prompts: await getAllQuestions() });
+  };
+
+  setHistory = () =>{
+    this.setState({history:[...this.state.history, this.state]})
+   this.props.setHistory()
+  }
+      
+  goBack = (event) =>{
+    event.preventDefault()
+    this.setState(this.state.history[this.state.history.length - 1])
+    this.props.goBack()
   }
 
   updateCurrentAnswers = (event, questionData) => {
     event.preventDefault();
     if (this.state.currentAnswers.includes(event.target.textContent)) {
-      const answersPerQuestion = [...this.state.currentAnswers]
-      const index = this.state.currentAnswers.indexOf(event.target.textContent)
-      answersPerQuestion.splice(index, 1)
-      return this.setState({currentAnswers: answersPerQuestion})
+      const answersPerQuestion = [...this.state.currentAnswers];
+      const index = this.state.currentAnswers.indexOf(event.target.textContent);
+      answersPerQuestion.splice(index, 1);
+      return this.setState({ currentAnswers: answersPerQuestion });
     }
-    if (+questionData.allowedResponses && this.state.currentAnswers.find(answer => questionData.choices.includes(answer))) {
-      let choiceToSwap = this.state.currentAnswers.find(answer => questionData.choices.includes(answer))
-      let indexToReplace = this.state.currentAnswers.indexOf(choiceToSwap)
-      let answerToReplace = this.state.currentAnswers
-      answerToReplace.splice(indexToReplace, 1, event.target.textContent)
+    if (
+      +questionData.allowedResponses &&
+      this.state.currentAnswers.find((answer) =>
+        questionData.choices.includes(answer)
+      )
+    ) {
+      let choiceToSwap = this.state.currentAnswers.find((answer) =>
+        questionData.choices.includes(answer)
+      );
+      let indexToReplace = this.state.currentAnswers.indexOf(choiceToSwap);
+      let answerToReplace = this.state.currentAnswers;
+      answerToReplace.splice(indexToReplace, 1, event.target.textContent);
       if (answerToReplace) {
-        return this.setState({currentAnswers: answerToReplace})
+        return this.setState({ currentAnswers: answerToReplace });
       }
     } else {
-      this.setState({currentAnswers: [...this.state.currentAnswers, event.target.textContent]})
-    }
-    this.props.updateActivityAnswers(event)
-    this.setState({currentAnswers: [...this.state.currentAnswers, event.target.textContent]})
+      this.setState({
```

Comment:
Does the linter suggest the breakup of the `setState` method? I feel like it takes up a lot of space when the one-line way is still readable (in my opinion).

Reply:
For some reason it does, but I am also not a fan of it, I much prefer the one line.